sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -216,11 +221,11 @@ public void startQueryJob(JobReference jobRef, JobConfigurationQuery queryConfig
     @Override
     public void startCopyJob(JobReference jobRef, JobConfigurationTableCopy copyConfig)
         throws IOException, InterruptedException {
-      Job job =
-          new Job()
-              .setJobReference(jobRef)
-              .setConfiguration(new JobConfiguration().setCopy(copyConfig));
-
+      Map<String, String> labelMap = new HashMap<>();
+      JobConfiguration config = new JobConfiguration();
+      config.setCopy(copyConfig);
+      config.setLabels(this.bqIOMetadata.addAdditionalJobLabels(labelMap));
+      Job job = new Job().setJobReference(jobRef).setConfiguration(config);
```

Comment:
I believe the protos might be immutable, and creating new instances when you call `setX` on them. You can do the whole thing fluently:

```suggestion
      Job job = new Job()
            .setJobReference(jobRef)
            .setConfiguration(new JobConfiguration()
                  .setCopy(copyConfig)
                  .setLabels(this.bqIOMetadata.addAdditionalJobLabels(labelMap)));
```

Reply:
Same for other occurrences in this file